test/images/agnhost/openidmetadata/openidmetadata.go

```diff
@@ -162,3 +168,36 @@ func withOAuth2Client(context.Context) (context.Context, error) {
 		})
 	return ctx, nil
 }
+
+// DNS can be available sometime after the container starts due to the way
+// networking is set up for Windows nodes with dockershim as the container runtime.
+// In this case the. we should make sure we are able to resolve the isser before
+// invoking oidc.NewProvider.
+// See https://github.com/kubernetes/kubernetes/issues/99470 for more details.
+func ensureWindowsDNSAvailability(issuer string) error {
+	if runtime.GOOS != "windows" {
```

Comment:
nit: I think it would be clearer to have this check outside this function surrounding the call ie:

```
if runtime.GOOS == "windows" {
  ensureWindowsDNSAvaliablity()
}